client/blocks/image-editor/index.jsx

```diff
@@ -72,14 +88,54 @@ const ImageEditor = React.createClass( {
 		} = this.props;
 
 		if ( newProps.media && ! isEqual( newProps.media, currentMedia ) ) {
+			this.checkForValidAspectRatioSettings();
+
 			this.props.resetAllImageEditorState();
 
 			this.updateFileInfo( newProps.media );
+
+			this.setDefaultAspectRatio();
 		}
 	},
 
 	componentDidMount() {
 		this.updateFileInfo( this.props.media );
+
+		this.setDefaultAspectRatio();
+	},
+
+	checkForValidAspectRatioSettings() {
+		const {
+			allowedAspectRatios,
+			defaultAspectRatio
+		} = this.props;
+
+		if (
+			allowedAspectRatios &&
+			allowedAspectRatios.length !== 1 &&
+			allowedAspectRatios.indexOf( defaultAspectRatio ) === -1
+		) {
+			// If allowedAspectRatios prop is specified and has more than one item,
+			// defaultAspectRatio prop must contain a value which is included in allowedAspectRatios.
+			throw new Error( 'Image Editor props: defaultAspectRatio not found in allowedAspectRatios' );
+		}
+	},
+
+	setDefaultAspectRatio() {
+		const {
+			defaultAspectRatio,
+			allowedAspectRatios
+		} = this.props;
+
+		if ( allowedAspectRatios && allowedAspectRatios.length === 1 ) {
+			this.props.setImageEditorAspectRatio( allowedAspectRatios[ 0 ] );
```

Comment:
> Do you think it's still a bug?

Not necessarily a bug, but more of a consistency issue. For example we could also just choose the first item of `allowedAspectRatios` as default, if  `defaultAspectRatio` is set to a nonsense value.